nx/lib/nx/serving.ex

```diff
@@ -616,8 +623,16 @@ defmodule Nx.Serving do
     batch = Nx.Batch.merge(batches)
     {:execute, function, module_state} = handle_batch(module, batch, module_state)
 
+    wrapped_function = fn ->
```

Comment:
The function is passed to `async_nolink` below, so this additional wrapper is the best solution I see.